### PR TITLE
Improve the design of the Guides page

### PR DESCRIPTION
**Pull request by alif-munim**
Signed-off-by: Alif Munim <alif.munim@ryerson.ca>

See previous PR, https://github.com/eclipse/codewind-docs/pull/638

## What type of PR is this ? 

- [ ] Bug fix
- [x] Enhancement

## What does this PR do ?
* Make guide titles blue to make their interactivity more apparent
* Add objective and image metadata to the guide markdown files
* Access and display objective and image metadata on each guide preview
* Remove "read more" link to eliminate redundancy
* Adjust spacing
* Adjust card layout for smaller screen sizes

## Which issue(s) does this PR fix ?
https://github.com/eclipse/codewind/issues/2929

## Does this PR require a documentation change ?
Not as far as I'm aware.

## Any special notes for your reviewer ?
Created a similar pull request previously but closed it as there were some validation errors (forgot sign-off statement on a commit).

### CODE REVIEW

**Comment by alif-munim** [comment]
@jcockbain integrated your feedback into this new PR. Please take a look when you get the chance

**Comment by alif-munim** [comment]
A look at the guide previews on smaller screen sizes:
image

File: docs/_layouts/guides.html [hunk 1]
```diff
@@ -13,24 +13,28 @@
...
-                        <div class="guides-post-content guides-content">
-                            <h2 class="guides-post-title"><a href="{{ post.url | relative_url }}"
-                                    class="guides-post-title-link">{{ post.summary_title | escape }}</a></h2>
-                            <p class="guides-post-date-mobile d-sm-block d-md-none">{{ post.date | date: date_format }}
-                            </p>
-                            <p class="guides-post-paragraph">
-                                {% if post.guide_description %}
-                                {{ post.guide_description }}
+                    <div class="media">
+                        <img src="{{ post.image }}" class="guide-image" alt="...">
```
Comment (jcockbain) [inline docs/_layouts/guides.html]: was there a reason for "..." as alt?

Comment (alif-munim) [reply docs/_layouts/guides.html]: I believe it's part of the bootstrap template. I'll fill that in now 👍

**Pull request merged**